homeassistant/components/iammeter/sensor.py

```diff
@@ -0,0 +1,130 @@
+"""Support for iammeter via local API."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import async_timeout
+from iammeter import real_time_api
+from iammeter.power_meter import IamMeterError
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import debounce
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 80
+DEFAULT_DEVICE_NAME = "IamMeter"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_DEVICE_NAME): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    }
+)
+
+SCAN_INTERVAL = timedelta(seconds=30)
+PLATFORM_TIMEOUT = 8
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Platform setup."""
+    config_host = config[CONF_HOST]
+    config_port = config[CONF_PORT]
+    config_name = config[CONF_NAME]
+    try:
+        with async_timeout.timeout(PLATFORM_TIMEOUT):
+            api = await real_time_api(config_host, config_port)
+    except asyncio.TimeoutError:
+        _LOGGER.error("Device is not ready")
+        raise PlatformNotReady
+
+    async def async_update_data():
+        try:
+            return await api.get_data()
+        except IamMeterError:
+            raise UpdateFailed
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=DEFAULT_DEVICE_NAME,
+        update_method=async_update_data,
+        update_interval=SCAN_INTERVAL,
+        request_refresh_debouncer=debounce.Debouncer(
+            hass, _LOGGER, cooldown=0.3, immediate=True
+        ),
+    )
+    await coordinator.async_refresh()
+    devices = []
+    for sensor_name, (row, idx, unit) in api.iammeter.sensor_map().items():
+        mac = api.iammeter.mac
+        serial_number = api.iammeter.serial_number
+        uid = f"{mac}-{serial_number}-{row}-{idx}"
```

Comment:
This seems a bit overkill. Aren't both mac and serial both unique IDs for a device? I would expect to be able to drop `mac`.

Reply:
thanks for your advice,I'v droped 'mac' in uid.